modules/stereo/src/descriptor.hpp

```diff
@@ -0,0 +1,441 @@
+        template <int step_start, int step_end, int step_inc, typename Kernel>
+        class CombinedDescriptor:public ParallelLoopBody
+        {
+        private:
+            uint8_t *image1, *image2;
+            int width, height,n2;
+            int *dst1, *dst2;
+            Kernel kernel_;
+            int n2_stop;
+        public:
+            CombinedDescriptor(int w, int h, int k2, int * distance1, int * distance2, Kernel kernel,int k2Stop) :
+                width(w), height(h), n2(k2),dst1(distance1), dst2(distance2), kernel_(kernel), n2_stop(k2Stop){}
+            void operator()(const cv::Range &r) const {
+                for (int i = r.start; i <= r.end ; i++)
+                {
+                    int rWidth = i * width;
+                    for (int j = n2 + 2; j <= width - n2 - 2; j++)
+                    {
+                        int c = 0;
+                        int c2 = 0;
+                        for(int step = step_start; step <= step_end; step += step_inc)
+                        {
+                            for (int ii = - n2; ii <= + n2_stop; ii += step)
+                            {
+                                int rrWidth = (ii + i) * width;
+                                int rrWidthC = (ii + i + n2) * width;
+                                for (int jj = j - n2; jj <= j + n2; jj += step)
+                                {
+                                    if (ii != i || jj != j)
+                                    {
+                                        kernel_(rrWidth,rrWidthC, rWidth, jj, j, c,c2);
+                                    }
+                                }
+                            }
+                        }
+                        dst1[rWidth + j] = c;
+                        dst2[rWidth + j] = c2;
+                    }
+                }
+            }
+        };
+        //!paralel implementation of the center symetric census
+        class parallelSymetricCensus:public ParallelLoopBody
+        {
+        private:
+            uint8_t *image1, *image2;
+            int *dst1, *dst2;
+            int n2, width, height, type;
+        public:
+            parallelSymetricCensus(uint8_t * img1, uint8_t * img2, int w, int h, int k2, int * distance1, int * distance2,const int t) :
+                image1(img1), image2(img2), dst1(distance1), dst2(distance2), n2(k2), width(w), height(h), type(t){}
+            void operator()(const cv::Range &r) const {
+                for (int i = r.start; i <= r.end ; i++)
+                {
+                    int distV = (i)* width;
+                    for (int j = n2; j <= width - n2; j++)
+                    {
+                        int c = 0;
+                        int c2 = 0;
+                        //the classic center symetric census which compares the curent pixel with its symetric not its center.
+                        for (int ii = -n2; ii < 0; ii++)
```

Comment:
I don't really see why symmetricCensus can't be implemented in terms of "kernels". It doesn't seem very different from everything else. Like, we know the coordinates of the center pixel, we know the "current pixel", we can easily evaluate the "symmetric" pixel. Am I missing something?